virttest/utils_test/libvirt.py

```diff
@@ -1541,6 +1541,93 @@ def create_redirdev_xml(redir_type="spicevmc", redir_bus="usb",
     return redir.xml
 
 
+def set_vm_attrs(vmxml, vm_attrs):
+    """
+    Set element/value pairs in VMXML instance
+
+    :param vmxml: VMXML instance of the domain
+    :param vm_attrs: dict of the attribute/value pairs in VMXML
+    :return the updated vmxml
+    """
+    for attr, value in list(vm_attrs.items()):
+        logging.debug('Set %s = %s', attr, value)
+        setattr(vmxml, attr, int(value) if value.isdigit() else value)
+    vmxml.xmltreefile.write()
+    vmxml.sync()
+    return vmxml
+
+
+def add_cpu_settings(vmxml, params):
+    """
+    Add cpu element/value pairs to the cpu xml section and keeping the
+    existing contents if they are already in the xml
+
+    :param vmxml: VMXML instance of the domain
+    :param params: dict of the cpu related parameter pairs
+    :return the updated vmxml
+    """
+    feature_list = None
+    if vmxml.xmltreefile.find('cpu'):
+        cpu_xml = vmxml.cpu
+        feature_list = cpu_xml.get_feature_list()
+    else:
+        cpu_xml = vm_xml.VMCPUXML()
+
+    if cpu_xml.xmltreefile.find('mode'):
+        cpu_mode = cpu_xml.mode
+    else:
+        cpu_mode = params.get("cpuxml_cpu_mode", "host-model")
+
+    if cpu_xml.xmltreefile.find("model"):
+        cpu_model = cpu_xml.model
+    else:
+        cpu_model = params.get("cpuxml_model", "")
+
+    if cpu_xml.xmltreefile.find("fallback"):
+        cpu_model = cpu_xml.fallback
+    else:
+        cpu_fallback = params.get("cpuxml_fallback", "forbid")
+
+    cpu_xml_new = vm_xml.VMCPUXML()
+    cpu_xml_new.xml = "<cpu><numa/></cpu>"
+    cpu_xml_new.mode = cpu_mode
+    cpu_xml_new.model = cpu_model
+    cpu_xml_new.fallback = cpu_fallback
+
+    feature_name_add = params.get("cpu_feature", None)
+    feature_policy_add = params.get("cpu_feature_policy", None)
+    if feature_list:
+        for i in range(0, len(feature_list)):
+            feature_name = cpu_xml.get_feature(i).get('name')
+            if feature_name == feature_name_add:
+                feature_name_add = ""
+            feature_policy = cpu_xml.get_feature(i).get('policy')
+            cpu_xml_new.add_feature(feature_name, feature_policy)
+    if feature_name_add and feature_policy_add:
+        cpu_xml_new.add_feature(feature_name_add, feature_policy_add)
+
+    if cpu_xml.xmltreefile.find('numa'):
+        cpu_xml_new = cpu_xml.get_numa_cell()
+    else:
+        cells = eval(params.get("cpuxml_numa_cell", "[]"))
+        for i in range(len(cells)):
+            numacell_xml = vm_xml.VMCPUXML().Numa_CellXML()
+            numacell_xml.update(cells[i])
+            cpu_xml_new.set_numa_cell(numacell_xml)
+
+        # Update the vcpu and memory values to match the cell config
```

Comment:
If only vcpu and memory values are needed to be set here, could you give some parameter samples in the comment line in order for better readability, or it is also ok to add it in the function docstring.